akka-cluster/src/test/scala/akka/cluster/protobuf/ClusterMessageSerializerSpec.scala

```diff
@@ -82,6 +82,21 @@ class ClusterMessageSerializerSpec extends AkkaSpec(
       checkSerialization(InternalClusterAction.Welcome(uniqueAddress, g2))
     }
 
+    "be compatible with wire format of version 2.5.9 (using Init singleton instead of class)" in {
+      // we must use the old singleton class name so that the other side will see an InitJoin
+      // but discard the config as it does not know about the config check
+      val oldClassName = "akka.cluster.InternalClusterAction$InitJoin$"
+      serializer.manifest(InternalClusterAction.InitJoin(ConfigFactory.empty())) should ===(oldClassName)
```

Comment:
I don't feel super confident with these kinds of tests that we really ensured compatibility again (and didn't break anything else). Could you create a test like the next one to make sure it still works at least for `InitJoin`?

Reply:
Ah, missed this one. The thing is that it is then serialized to empty byte array, so that means that it is just about the class name passed to `fromBinary` which is tested on the rows below.

I should add one for the other half of the issue with the ack just being sent as address though.